tests/drivers/mspi/error_cases/src/main.c

```diff
@@ -0,0 +1,686 @@
+
+	dev_cfg.data_rate = MSPI_DATA_RATE_DUAL;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_DATA_RATE, &dev_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_14a_MSPI_DEVICE_CONFIG_CPP_eq_MSPI_CPP_MODE_0)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.cpp = MSPI_CPP_MODE_0;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CPP, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_14b_MSPI_DEVICE_CONFIG_CPP_eq_MSPI_CPP_MODE_1)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.cpp = MSPI_CPP_MODE_1;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CPP, &dev_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_14c_MSPI_DEVICE_CONFIG_CPP_eq_MSPI_CPP_MODE_2)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.cpp = MSPI_CPP_MODE_2;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CPP, &dev_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_14d_MSPI_DEVICE_CONFIG_CPP_eq_MSPI_CPP_MODE_3)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.cpp = MSPI_CPP_MODE_3;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CPP, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_15a_MSPI_DEVICE_CONFIG_ENDIAN_eq_MSPI_XFER_LITTLE_ENDIAN)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.endian = MSPI_XFER_LITTLE_ENDIAN;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_ENDIAN, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_15b_MSPI_DEVICE_CONFIG_ENDIAN_eq_MSPI_XFER_BIG_ENDIAN)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.endian = MSPI_XFER_BIG_ENDIAN;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_ENDIAN, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_16a_MSPI_DEVICE_CONFIG_CE_POL_eq_MSPI_CE_ACTIVE_LOW)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.ce_polarity = MSPI_CE_ACTIVE_LOW;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CE_POL, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_16b_MSPI_DEVICE_CONFIG_CE_POL_eq_MSPI_CE_ACTIVE_HIGH)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.ce_polarity = MSPI_CE_ACTIVE_HIGH;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_CE_POL, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_17a_MSPI_DEVICE_CONFIG_DQS_true)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.dqs_enable = true;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_DQS, &dev_cfg, -ENOTSUP);
+}
+
+ZTEST(mspi_error_cases, test_17b_MSPI_DEVICE_CONFIG_DQS_false)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.dqs_enable = false;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_DQS, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_18a_MSPI_DEVICE_CONFIG_RX_DUMMY_MAX)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.rx_dummy = UINT16_MAX;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_RX_DUMMY, &dev_cfg, 0);
+}
+
+ZTEST(mspi_error_cases, test_19a_MSPI_DEVICE_CONFIG_TX_DUMMY_MAX)
+{
+	struct mspi_dev_cfg dev_cfg = device_cfg[0];
+
+	dev_cfg.tx_dummy = UINT16_MAX;
+
+	check_mspi_dev_config(MSPI_DEVICE_CONFIG_TX_DUMMY, &dev_cfg, 0);
```

Comment:
```suggestion
	dev_cfg.rx_dummy = UINT16_MAX;

	check_mspi_dev_config(MSPI_DEVICE_CONFIG_RX_DUMMY, &dev_cfg, -ENOTSUP);
}

ZTEST(mspi_error_cases, test_19a_MSPI_DEVICE_CONFIG_TX_DUMMY_MAX)
{
	struct mspi_dev_cfg dev_cfg = device_cfg[0];

	dev_cfg.tx_dummy = UINT16_MAX;

	check_mspi_dev_config(MSPI_DEVICE_CONFIG_TX_DUMMY, &dev_cfg, -ENOTSUP);
```
In the flpr application I saw an assert for the number of shifts greater than 63.
@mif1-nordic or @magp-nordic Do we have any restrictions on the number of dummy clocks? Could you please confirm?